feat(monitoring-connector) add documentation for Datacore new plugin … (#3107)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/hardware-storage-datacore-api.md

---
id: hardware-storage-datacore-api
title: Datacore RestApi
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Pack assets

### Templates

The Monitoring Connector **Datacore Sansymphony RESTAPI** brings a host template:

* **HW-Storage-Datacore-Sansymphony-RESTAPI-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="HW-Storage-Datacore-Sansymphony-RESTAPI-custom" label="HW-Storage-Datacore-Sansymphony-RESTAPI-custom">

| Service Alias   | Service Template                                              | Service Description                                     |
|:----------------|:--------------------------------------------------------------|:--------------------------------------------------------|
| Alerts          | HW-Storage-Datacore-Sansymphony-Alerts-RESTAPI-custom         | Check alerts                                            |

> | Monitors Status | HW-Storage-Datacore-Sansymphony-Status-Monitor-RESTAPI-custom | Check Datacore "monitors" object filtered by parameters |

Reviewer: ```suggestion
| Monitors-Status | HW-Storage-Datacore-Sansymphony-Status-Monitor-RESTAPI-custom | Monitor Datacore "monitors" object. |
```

> If **Discovery** is checked, it means a service discovery rule exists for this service template.

</TabItem>
</Tabs>

### Discovery rules

#### Service discovery

| Rule name                         | Description                                                |
|:----------------------------------|:-----------------------------------------------------------|
| Datacore-Sansymphony-Restapi-Pool | Discover the IDs of the pools from your Datacore Sansymphony Rest API. |

More information about discovering services automatically is available on the [dedicated page](/docs/monitoring/discovery/services-discovery)
and in the [following chapter](/docs/monitoring/discovery/services-discovery/#discovery-rules).

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics linked to each service.

<Tabs groupId="sync">
<TabItem value="Alerts" label="Alerts">

| Metric name                   | Unit  |
|:------------------------------|:------|
| datacore.event.error.count    | count |
| datacore.alerts.warning.count | count |
| datacore.alerts.info.count    | count |
| datacore.alerts.trace.count   | count |

</TabItem>

> <TabItem value="Monitors Status" label="Monitors Status">

Reviewer: ```suggestion
<TabItem value="Monitors-Status" label="Monitors-Status">
```

| Metric name    | Unit  |
|:---------------|:------|
| *global*#state | N/A   |

</TabItem>
<TabItem value="Pool" label="Pool">

| Metric name                        | Unit  |
|:-----------------------------------|:------|
| datacore.pool.oversubscribed.bytes | bytes |

</TabItem>
</Tabs>

## Prerequisites

> Configure Datacore Rest Api : https://docs.datacore.com/RESTSupport-WebHelp/RESTSupport-WebHelp/Installation_Instructions.htm

Reviewer: ```suggestion
To be able to monitor Datacore SANsymphony, you must configure an [access to the REST API](https://docs.datacore.com/RESTSupport-WebHelp/RESTSupport-WebHelp/Installation_Instructions.htm).
```

---
id: hardware-storage-datacore-api
title: Datacore RestApi
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Pack assets

### Templates

The Monitoring Connector **Datacore Sansymphony RESTAPI** brings a host template:

* **HW-Storage-Datacore-Sansymphony-RESTAPI-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="HW-Storage-Datacore-Sansymphony-RESTAPI-custom" label="HW-Storage-Datacore-Sansymphony-RESTAPI-custom">

| Service Alias   | Service Template                                              | Service Description                   |
|:----------------|:--------------------------------------------------------------|:--------------------------------------|
| Alerts          | HW-Storage-Datacore-Sansymphony-Alerts-RESTAPI-custom         | Check alerts                          |

> | Monitors Status | HW-Storage-Datacore-Sansymphony-Status-Monitor-RESTAPI-custom | Check elements filtered by parameters |

Reviewer 2: unclear

> The services listed above are created automatically when the **HW-Storage-Datacore-Sansymphony-RESTAPI-custom** host template is used.

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias | Service Template                                     | Service Description | Discovery  |
|:--------------|:-----------------------------------------------------|:--------------------|:----------:|
| Pool          | HW-Storage-Datacore-Sansymphony-Pools-RESTAPI-custom | Check pool usage    | X          |

> The services listed above are not created automatically when a host template is applied. To use them, [create a service manually](/docs/monitoring/basic-objects/services), then apply the service template you want.

> If **Discovery** is checked, it means a service discovery rule exists for this service template.

</TabItem>
</Tabs>

### Discovery rules

#### Service discovery

| Rule name                         | Description                                                |
|:----------------------------------|:-----------------------------------------------------------|

> | Datacore-Sansymphony-Restapi-Pool | discover pools id from your Datacore Sansymphony rest api. |

Reviewer 2: ```suggestion
| Datacore-Sansymphony-Restapi-Pool | Discover the IDs of the pools from your Datacore Sansymphony Rest API. |
```

More information about discovering services automatically is available on the [dedicated page](/docs/monitoring/discovery/services-discovery)
and in the [following chapter](/docs/monitoring/discovery/services-discovery/#discovery-rules).

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics linked to each service.

<Tabs groupId="sync">
<TabItem value="Alerts" label="Alerts">

| Metric name                   | Unit  |
|:------------------------------|:------|
| datacore.event.error.count    | count |
| datacore.alerts.warning.count | count |
| datacore.alerts.info.count    | count |
| datacore.alerts.trace.count   | count |

</TabItem>
<TabItem value="Monitors Status" label="Monitors Status">

| Metric name    | Unit  |
|:---------------|:------|
| *global*#state | N/A   |

</TabItem>
<TabItem value="Pool" label="Pool">

| Metric name                        | Unit  |
|:-----------------------------------|:------|
| datacore.pool.oversubscribed.bytes | bytes |

</TabItem>
</Tabs>

## Prerequisites

> 

Reviewer 2: sample text

1. If you have used a host template and checked **Create Services linked to the Template too**, the services linked to the template have been created automatically, using the corresponding service templates. Otherwise, [create manually the services you want](/docs/monitoring/basic-objects/services) and apply a service template to them.
2. Fill in the macros you want (e.g. to change the thresholds for the alerts). Some macros are mandatory (see the table below).

<Tabs groupId="sync">
<TabItem value="Alerts" label="Alerts">

| Macro           | Description                                                                                                                                                    | Default value     | Mandatory   |
|:----------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|
| MAXALERTAGE     | filter alerts to check those newer than this parameter (s)                                                                                                     | 86400             |             |

> | FILTERSERVER    | Define which devices sh ould be monitored based on the MachineName. This option willbe treated as a regular expression. By default all machine will be checked |                   |             |

Reviewer 2: ```suggestion
| FILTERSERVER    | Define which devices should be monitored based on the MachineName. This option will be treated as a regular expression. By default all machines will be checked. |                   |             |
```

Reviewer 2: same comment as in FR

> | WARNINGWARNING  | thresholds                                                                                                                                                      |                   |             |

Reviewer 2: Capital T for all instances of "threshold" (and it should be threshold in the singular, shouldn't it?)
